go/cmd/server/logging/logging.go

```diff
@@ -0,0 +1,306 @@
+package logging
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/apache/arrow/go/v8/arrow"
+	"github.com/apache/arrow/go/v8/arrow/array"
+	"github.com/apache/arrow/go/v8/arrow/memory"
+	"github.com/feast-dev/feast/go/internal/feast"
+	"github.com/feast-dev/feast/go/internal/feast/model"
+	"github.com/feast-dev/feast/go/protos/feast/serving"
+	"github.com/feast-dev/feast/go/protos/feast/types"
+	gotypes "github.com/feast-dev/feast/go/types"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type Log struct {
+	// Example: val{int64_val: 5017}, val{int64_val: 1003}
+	EntityValue []*types.Value
+
+	FeatureValues   []*types.Value
+	FeatureStatuses []serving.FieldStatus
+	EventTimestamps []*timestamppb.Timestamp
+	RequestContext  map[string]*types.RepeatedValue
+}
+
+type MemoryBuffer struct {
+	featureService *model.FeatureService
+	logs           []*Log
+}
+
+type LoggingService struct {
+	memoryBuffer      *MemoryBuffer
+	logChannel        chan *Log
+	fs                *feast.FeatureStore
+	offlineLogStorage OfflineLogStorage
+}
+
+func NewLoggingService(fs *feast.FeatureStore, logChannelCapacity int, featureServiceName string, enableLogging bool) (*LoggingService, error) {
```

Comment:
`enabledLogging` seems very strange parameter in logging service constructor.
I understand what's the goal here, so I think a better approach would be to have `DummyLoggingService` (separate type) with separate constructor `NewDummyLoggingService` that would have method `EmitLog` that does nothing

Reply:
I think that in itself may be redundant since we are essentially just copy pasting code for the dummy log service.
Also emit log is not the part that I disable. I disable the log processing part which I think makes sense even for log service.